### ISSUE TITLE
sparse_categorical_crossentropy does not use underlying logits of tf.keras.layers.Softmax

### ISSUE THREAD

**Issue by mpdn** [tensorflow/tensorflow#32869]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.0.0-rc0
- Python version: 3.6

**Describe the current behavior**
As per the comment in `sparse_categorical_crossentropy`, the function will use the underlying logits for better numerical stability if the last op is a softmax: https://github.com/tensorflow/tensorflow/blob/1ae01a07f5e4b2911218d5fc3419b3eaed48b75b/tensorflow/python/keras/backend.py#L4522-L4533

But since keras adds an identity op to the output of every layer, the last node of a layer will always be `Identity` so the condition in the above if statement will never fail for keras models and the underlying logits are never used: https://github.com/tensorflow/tensorflow/blob/b0aa37c3fdff00b5c69bce11c716c3e56f656dd4/tensorflow/python/keras/engine/base_layer.py#L851-L853

Using non-keras-layers like `tf.nn.softmax` will work, however.

**Describe the expected behavior**
Instead, `sparse_categorical_crossentropy` should check the first non-`Identity` node when determining whether the last node is a softmax node. I.e. it should first walk the chain of identity nodes.

**Code to reproduce the issue**
```
import tensorflow as tf

x = tf.keras.layers.Input(shape=[10])

softmax_keras = tf.keras.layers.Softmax()(x)
softmax_tf = tf.nn.softmax(x)

y_true = tf.keras.layers.Input(shape=[])
loss_keras = tf.keras.losses.sparse_categorical_crossentropy(y_true, softmax_keras)
loss_tf = tf.keras.losses.sparse_categorical_crossentropy(y_true, softmax_tf)

def transitive_inputs(tensor):
    return set([tensor.op.type] + [inp for tensor in tensor.op.inputs for inp in transitive_inputs(tensor)])

print('Softmax' in transitive_inputs(loss_keras)) # prints "True"
print('Softmax' in transitive_inputs(loss_tf)) # prints "False"
```

**Comment by noctune** [comment]
This seems to have an absolutely HUGE impact on my models. I *really* recommend anyone using `tf.keras.layers.Softmax` to use `tf.nn.softmax` instead until this issue is fixed.

**Comment by oanush** [comment]
Tried replicating with given code for TF-2.0rc2, the same error was received. Kindly find the [gist](https://colab.sandbox.google.com/gist/oanush/78bd4585819aa58fbc13155a1ab3871d/32869.ipynb) of colab.Thanks!

**Comment by jvishnuvardhan** [comment]
@mpdn I think this was resolved in `tf-nightly`. Please check the [gist here](https://colab.sandbox.google.com/gist/jvishnuvardhan/d30cfe2f0b6a5c3f6b72227378fbdad9/32869.ipynb).

I am closing this issue as it was resolved. Please feel free to reopen if the issue persists again. Thanks!
